### DISCUSSION TITLE
Running a group account without sharing a password

### DISCUSSION THREAD

**Discussion by linux-lukas** [mastodon/mastodon#19287] [General]
### A. Problem / Goal

Currently it is only possible to operate accounts by sharing the password. However, this is for reasons of IT security or the fact that people behind the account should only be able to do limited things. But that is currently not possible, neither with Mastodon, nor with any other Fediverse service, as far as I know.

The goal would be to implement a concept in Mastodon that makes it possible to control users without the password, in that the account owner, who knows the password, authorises users to perform activities (read, write, all process) on behalf of the account owner using the trade (so to speak "@user@domain.tld on behalf of @gruppenaccount@domain.tld"). However, it would be important to consider how this concept can be implemented in such a way that it can be standardised in ActivityPub so that all other Fediverse services can also benefit from it - the interoperability of this function with the other Fediverse services should also be actively considered. 

# B. Considered solution 
Implement finely granular rights management. 

### C. Alternatives
This would entail the sharing of passwords and therefore potential risks. 

### D. Effort

The effort is likely to be great, since a fine-grained assignment of rights would have to be implemented.

**Comment by linux-lukas** [comment]
It would then perhaps also be important that accounts of authorities, governments or companies, such as Dr. Oetker Pizza DE or DB Cargo, are automatically appended with endings such as "/tn", so that they are traceable.

**Comment by ghost** [comment]
There is a similar, yet restricted feature in Friendica. There you can "delegate" an account to other accounts on the same instance. It is not possible (yet?) to delegate Friendica accounts across instances. However, I would like to see this being possible.

This discussion is related to:
https://github.com/mastodon/mastodon/issues/1405
https://github.com/mastodon/mastodon/issues/12153
